libs/interpret-vision/src/lib/VisionExplanationDashboard/utils/FlyoutUtils.tsx

```diff
@@ -34,13 +34,14 @@ export function onRenderCell(
 
 export function generateSelectableObjectDetectionIndexes(
   prefix: string,
-  item: IVisionListItem | undefined
+  item: IVisionListItem | undefined,
+  class_names: string[] | undefined
 ): IComboBoxOption[] {
   const temp = item?.odPredictedY;
   const selectableObjectIndexes: IComboBoxOption[] = [];
-  if (temp) {
+  if (temp && class_names) {
     for (let i = 0; i < Object.values(temp).length; i++) {
-      const className = item?.predictedY[i];
+      const className = class_names[temp[i][0] - 1];
```

Comment:
class_names is an optional parameter so maybe might be good to fallback to old logic if it is undefined?